### PR TITLE
TerraFirmaCraft Food Compatibility for MetaItems

### PR DESCRIPTION
**Pull request by bruberu**
Allows for MetaItem food to give hunger and saturation as usual with TerraFirmaCraft installed, as well as have nutrition and heating properties.

This is done through the ITFCFoodComponent interface, which gives TFC's CapabilityFood to ItemStacks with this component. Implementations of this are expected to include a call to player.getFoodStats().addStats(null, stack) within their onFoodEaten() overrides, so that TFC can access the capability.

Add-ons (such as my own GTFO) will have to either implement this interface, or extend its default implementation, TFCFoodComponent, when creating their own FoodStats.

### CODE REVIEW

File: src/main/java/gregification/tfc/TFCModule.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+
+@Module.Root(name = "Gregification: Ex TerraFirmaCraft")
```
Comment (DStrand1) [inline src/main/java/gregification/tfc/TFCModule.java:37]: `Ex` here I think is a copy-paste mistake

File: src/main/java/gregification/tfc/TFCModule.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+public class TFCModule implements Module {
+    @Module.Log
+    public static Logger logger;
+
+    @Override
+    public boolean isModuleActive() {
+        return Loader.isModLoaded(ModIDs.MODID_TFC);
```
Comment (DStrand1) [inline src/main/java/gregification/tfc/TFCModule.java:44]: There should be a top-level config in `BaseConfig` that enables/disables the entire module, and is checked here

**Review by DStrand1 — COMMENTED** [review]
Looks pretty good to me.

You mentioned adding additional stats in GTFO. We need to be careful to avoid circular dependencies in addons, so I think any stats that you want for GTFO foods should be PR'd here (or else this work needs to move to CEu itself)

File: src/main/java/gregification/base/BaseConfig.java [hunk 3]
```diff
@@ -19,4 +19,8 @@ public class BaseConfig {
     @Comment({"Enable Open Computers integration.", "Default: true"})
     @RequiresMcRestart
     public static boolean enableOpenComputersModule = true;
+
+    @Comment({"Enable Open Computers integration.", "Default: true"})
```
Comment (DStrand1) [inline src/main/java/gregification/base/BaseConfig.java:23]: Comment is wrong

**Review by DStrand1 — APPROVED** [review]
LGTM, though sorry to potentially change my mind again regarding GTFO compat.

If you want to leave it like this, that's totally okay, though if you want direct control over it in GTFO, you could set up an InterModComms channel for gregification to listen on for GTFO to add the stats by itself. Up to you how important controlling this in GTFO is (arguably its better, since you can add food with TFC stats without forcing a Gregification update, though I'll leave that decision to you). If you want help setting up IMC, I can make the Gregification side and write a comment on how to do it from GTFO's side

**Comment by bruberu** [comment]
Sorry to get back so late; I think an IMC setup would probably work much better (especially given how it's hard to make a PR here every time I want to update nutrition stats for GTFO items). I'll try my hand at putting in the IMC myself.

**Pull request merged**